### PR TITLE
[EBS - 133 ] - WEB | Convert Col component to Typescript

### PR DESCRIPTION
**Pull request by hamzahsn**
**Description:**

This PR is about converting Row component to typescript

**AC:**

**out of scope:**

**Jira Link:**

https://uvitadev.atlassian.net/browse/EBS-133

**Screenshot:**

### CODE REVIEW

File: src/Grid/Col/index.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import React, { FC, CSSProperties, ReactNode } from "react";
+import classNames from "../../utils/classNames";
+
+type ColProps = {
+  children?: ReactNode;
```
Comment (a7madgamal) [inline src/Grid/Col/index.tsx:5]: no need to type children unless we want to require it then lets remove the `?`

File: src/Grid/Col/index.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  lg?: 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | "hide";
+  md?: 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | "hide";
+  sm?: 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | "hide";
+  xs?: 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | "hide";
+  offset?: 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11;
+  offsetMd?: 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11;
+  offsetSm?: 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11;
+  offsetXs?: 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11;
+  style?: CSSProperties;
+  center?: boolean;
```
Comment (a7madgamal) [inline src/Grid/Col/index.tsx:14]: we don't use center lets remove it

Comment (hamzahsn) [reply src/Grid/Col/index.tsx:14]: but `center` is used in storybook as a prop for `Col` component and that's why I typed it here

Comment (a7madgamal) [reply src/Grid/Col/index.tsx:14]: you should remove it from there instead of adding it here without use

Comment (a7madgamal) [reply src/Grid/Col/index.tsx:14]: or implement the center logic

Comment (hamzahsn) [reply src/Grid/Col/index.tsx:14]: removing it from here and convert `Grid.jsx` to `Grid.tsx` will lead to a crush for the app cause center prop is not typed

Comment (hamzahsn) [reply src/Grid/Col/index.tsx:14]: center logic ? 🤔

Comment (a7madgamal) [reply src/Grid/Col/index.tsx:14]: thats why you should do one pr at a time.

File: src/Grid/Col/index.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+};
+
+const Col: FC<ColProps> = ({
+  children,
+  lg = 12,
+  md = null,
```
Comment (a7madgamal) [inline src/Grid/Col/index.tsx:19]: please remove all `null`s

**Comment by a7madgamal** [comment]
@hamzahsn @Weetbix I fixed eslint job and tested that it catches errors in correct files
had to use temporary ignore comments to avoid this pr getting bigger

**Pull request merged**